### PR TITLE
vault: Allow to provide extra statement for roles creation.

### PR DESCRIPTION
**Pull request by cyrilgdn**

### CODE REVIEW

**Comment by greut** [comment]
Can you point us where you're using that? Out of curiosity.

**Comment by cyrilgdn** [comment]
> Can you point us where you're using that? Out of curiosity.

Yes, here: https://github.com/edgelaboratories/stacks/pull/2674

File: vault.tf [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
-  creation_statements = [
-    "CREATE USER '{{name}}'@'%' IDENTIFIED BY '{{password}}';",
-    "GRANT ${each.value} ON ${var.database}.* TO '{{name}}'@'%';",
-  ]
+  creation_statements = concat(
+    [
+      "CREATE USER '{{name}}'@'%' IDENTIFIED BY '{{password}}';",
+      "GRANT ${each.value} ON ${var.database}.* TO '{{name}}'@'%';",
+    ], var.roles_extra_statements[each.key] != null ? var.roles_extra_statements[each.key] : [],
```
Comment (greut) [inline vault.tf:19]: This looks identical.

```suggestion
    ], coalesce(var.roles_extra_statements[each.key], []),
```

**Comment by cyrilgdn** [comment]
> What is missing is some documentation in the README. +1

@greut Added, I also added a `vault_` prefix in the variable name like all other Vault variables.

File: README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ In particular:
 
 ```hcl
 module "my_database" {
-  source = "git@github.com:edgelaboratories/terraform-mysql-db?ref=v0.1.2"
+  source = "git@github.com:edgelaboratories/terraform-mysql-db?ref=v0.1.3"
```
Comment (greut) [inline README.md:20]: It's a new feature more than a bugfix.

```suggestion
  source = "git@github.com:edgelaboratories/terraform-mysql-db?ref=v0.2.0"
```

**Review by greut — APPROVED** [review]
Go 0.2.0!

**Pull request merged**